compiler/rustc_codegen_ssa/src/base.rs

```diff
@@ -898,55 +894,6 @@ impl CrateInfo {
             }
         }
 
-        // Handle circular dependencies in the standard library.
-        // See comment before `add_linked_symbol_object` function for the details.
-        // If global LTO is enabled then almost everything (*) is glued into a single object file,
-        // so this logic is not necessary and can cause issues on some targets (due to weak lang
-        // item symbols being "privatized" to that object file), so we disable it.
-        // (*) Native libs, and `#[compiler_builtins]` and `#[no_builtins]` crates are not glued,
-        // and we assume that they cannot define weak lang items. This is not currently enforced
-        // by the compiler, but that's ok because all this stuff is unstable anyway.
```

Comment:
This logic is mostly unrelated to `no_builtins`, it should not be removed.
The only thing that should need an update here is this comment.

"Native libs are not glued, and we assume that they cannot define weak lang items."

Reply:
Thanks, I've updated it.